src/plugins/auto/auto_schedule.cpp

```diff
@@ -217,107 +237,178 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
             _loadContext[ACTUALDEVICE].deviceInfo = validDevices.front();
             _loadContext[ACTUALDEVICE].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] =
                 IE::PluginConfigParams::THROUGHPUT;
-            isCumulative = false;
-        } else {
-            // When the hint is ctput and there are more than one device, the MULTI logic is used
-            std::string deviceName = "MULTI:";
+        } else if (validDevices.size() > 1) {
+            _loadContext[ACTUALDEVICE].isEnabled = false;
+            _autoSContext->_devicePriorities.clear();
+            std::copy(std::begin(validDevices),
+                      std::end(validDevices),
+                      std::back_inserter(_autoSContext->_devicePriorities));
+            // Total number of devices in CTPUT
+            _nCTputDeviceNums = validDevices.size();
+            DeviceInformation cpuDeviceInformation;
+            const auto CPUIter =
+                std::find_if(validDevices.begin(), validDevices.end(), [](const DeviceInformation& d) -> bool {
+                    return d.deviceName.find("CPU") != std::string::npos;
+                });
+            if (CPUIter != validDevices.end()) {
+                // no user set affinity then set CORE to CPU
+                CPUIter->config.insert({ov::affinity.name(), ov::Any(ov::Affinity::CORE).as<std::string>()});
+                cpuDeviceInformation = *CPUIter;
+                validDevices.erase(CPUIter);
+            }
+            // Generate contexts for loading each device
+            _pCTPUTLoadContext.reset(new AutoLoadContext[_nCTputDeviceNums]);
+            int idx = 0;
             for (auto& device : validDevices) {
-                deviceName += device.deviceName;
-                deviceName += ((device.deviceName == validDevices.back().deviceName) ? "" : ",");
+                _pCTPUTLoadContext[idx].deviceInfo = device;
+                _pCTPUTLoadContext[idx].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] = IE::PluginConfigParams::THROUGHPUT;
+                idx++;
+            }
+            // If there is a CPU, the CPU is loaded last
+            if (cpuDeviceInformation.deviceName.find("CPU") != std::string::npos) {
+                _pCTPUTLoadContext[idx].deviceInfo = cpuDeviceInformation;
+                _pCTPUTLoadContext[idx].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] = IE::PluginConfigParams::THROUGHPUT;
             }
-            _loadContext[ACTUALDEVICE].deviceInfo.deviceName = deviceName;
-            _loadContext[ACTUALDEVICE].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] =
-                InferenceEngine::PluginConfigParams::CUMULATIVE_THROUGHPUT;
-            _loadContext[ACTUALDEVICE].deviceInfo.config[CONFIG_KEY(PERF_COUNT)] =
-                _autoSContext->_needPerfCounters ? InferenceEngine::PluginConfigParams::YES
-                                                 : InferenceEngine::PluginConfigParams::NO;
-            if (_autoSContext->_bindBuffer)
```

Comment:
can we add bind schedule support for auto ctput? or do you plan in a separate PR?